shared/src/fs.rs

```diff
@@ -18,3 +20,25 @@ where
         .and_then(Iterator::collect::<std::io::Result<Vec<_>>>)
         .map(|dir_entries| dir_entries.into_iter().map(|dir_entry| dir_entry.path()))
 }
+
+#[cfg(unix)]
+pub fn is_executable<P: AsRef<Path>>(path: P) -> bool {
+    use std::os::unix::fs::PermissionsExt;
+
+    path.as_ref()
+        .metadata()
+        .map(|metadata| metadata.is_file() && metadata.permissions().mode() & 0o111 != 0)
+        .ok()
+        .unwrap_or_default()
+}
+
+#[cfg(unix)]
+#[allow(clippy::missing_errors_doc)]
```

Comment:
(Optional)

I think it would be fine to disable this lint globally in the shared crates if you preferred, since it's disabled a few times already:
https://github.com/search?q=repo%3Aheroku%2Fbuildpacks-jvm+missing_errors_doc&type=code